goal-planning/goal-planning-frontend
# Feat / Registration Form Creating User Account

## Changes
1. Configure redux async thunk logic to create user account at server register endpoint.
2. Created register form HTML.
3. Configure express routes to register and utilize json response.

## Purpose
Create a working logic registration form.

## Approach
Setup redux async thunk logic to await server registration response and apply some behavioral UI to test registration success like navigating user to login page. (Still needs more polishing like email confirmation and application snack bar notifications)

## Testing Steps
1. Run the sever application by opening backend code on VsCode. Open VsCode terminal and run `pipenv shell` followed by `cd app/` followed by `python manage.py runserver`.
2. Run the frontend application by opening frontend code on VsCode. Open VsCode terminal and run `cd frontend/` followed by `npm install` followed by `npm run start`.
3. Open the frontend code at `http://localhost:3000/register` to view registration form.
4. Fill in registration form and submit. Confirm you are routed to the `Login` page.
5. Check on the server `http://127.0.0.1:8000/admin/users/useraccount/` to confirm user account is created.
